### PR TITLE
[Merged by Bors] - chore: add some `fun_prop` attributes for continuity

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousOn.lean [hunk 1]
```diff
@@ -978,6 +978,13 @@ theorem Continuous.comp_continuousOn {g : β → γ} {f : α → β} {s : Set α
   hg.continuousOn.comp hf (mapsTo_univ _ _)
 #align continuous.comp_continuous_on Continuous.comp_continuousOn
 
+@[fun_prop]
+theorem Continuous.comp_continuousOn'
```
Comment (lecopivo) [inline Mathlib/Topology/ContinuousOn.lean:982]: I just realized that `fun_prop` stores only one composition theorem for each function property. This is true for any lambda theorem (_id,_const,_comp,_pi,_apply). Thus this this theorem clashes with `ContinuousOn.comp''` and `ContinuousOn.comp'`.

The current behavior is that only the latest lambda theorem is used. I think it is ok to keep it marked as `fun_prop` for now. I will add a support for multiple variants of lambda theorems, I already have this working for `fun_trans` so it should be quite easy.

**Comment by lecopivo** [comment]
The rest of the theorems should be fine.

**Review by YaelDillies — APPROVED** [review]
If Tomáš approves...

maintainer merge

File: Mathlib/Topology/Algebra/Polynomial.lean [hunk 2]
```diff
@@ -23,8 +23,8 @@ In this file we prove the following lemmas.
 * `Polynomial.continuous`:  `Polynomial.eval` defines a continuous functions;
   we also prove convenience lemmas `Polynomial.continuousAt`, `Polynomial.continuousWithinAt`,
   `Polynomial.continuousOn`.
-* `Polynomial.tendsto_norm_atTop`: `fun x ↦‖Polynomial.eval (z x) p‖` tends to infinity
-  provided that `fun x ↦ ‖z x‖` tends to infinity and `0 < degree p`;
+* `Polynomial.tendsto_norm_atTop`: `λ x, ‖Polynomial.eval (z x) p‖` tends to infinity provided that
+  `fun x ↦ ‖z x‖` tends to infinity and `0 < degree p`;
```
Comment (dupuisf) [inline Mathlib/Topology/Algebra/Polynomial.lean:27]:
```suggestion
* `Polynomial.tendsto_norm_atTop`: `fun x ↦ ‖Polynomial.eval (z x) p‖` tends to infinity provided that
  `fun x ↦ ‖z x‖` tends to infinity and `0 < degree p`;
```
Aren't we moving towards `fun` as the standard?

**Review by dupuisf — COMMENTED** [review]
bors d+

File: Mathlib/Topology/Algebra/Polynomial.lean [hunk 3]
```diff
@@ -23,8 +23,8 @@ In this file we prove the following lemmas.
 * `Polynomial.continuous`:  `Polynomial.eval` defines a continuous functions;
   we also prove convenience lemmas `Polynomial.continuousAt`, `Polynomial.continuousWithinAt`,
   `Polynomial.continuousOn`.
-* `Polynomial.tendsto_norm_atTop`: `fun x ↦‖Polynomial.eval (z x) p‖` tends to infinity
-  provided that `fun x ↦ ‖z x‖` tends to infinity and `0 < degree p`;
+* `Polynomial.tendsto_norm_atTop`: `fun x ↦ ‖Polynomial.eval (z x) p‖` tends to infinity provided that
+  `fun x ↦ ‖z x‖` tends to infinity and `0 < degree p`;
```
Comment (YaelDillies) [inline Mathlib/Topology/Algebra/Polynomial.lean:27]: Correctly rewrapped:
```suggestion
* `Polynomial.tendsto_norm_atTop`: `fun x ↦ ‖Polynomial.eval (z x) p‖` tends to infinity
  provided that `fun x ↦ ‖z x‖` tends to infinity and `0 < degree p`;
```

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**